Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -43,6 +44,13 @@ private void AddUnstickVerb(EntityUid uid, StickyComponent component, GetVerbsEv
         if (component.StuckTo == null || !component.CanUnstick)
             return;
 
+        // we can't use args.CanAccess, because stuck in another container
+        // we also need to ignore entity that it stuck to
+        var inRange = _interactionSystem.InRangeUnobstructed(uid, args.User,
+            predicate: entity => component.StuckTo == entity);
+        if (!inRange || !args.CanInteract)
```

Comment:
If it'd work the same, then put the `!args.CanInteract` check before the `inRange` check to exit early so that it doesn't need to raycast. So maybe up in the above if statement.